Update reference to secret for usercatalogue api key

norrs opened the pull request.

norrs committed:
Update reference to secret for usercatalogue api key

norrs commented:
This was done on friday 18 september when me and @svenngjeto was rolling out sandbox. this key is hardcoded, probably not the way we want to do it ...

norrs commented:
But for now this is good enough @svenngjeto , we should make a task for fixing this properly so we can have unique API keys across environments (sandbox, test, prod etc)

norrs commented:
Hm, we probably need this task anyway and solve it next sprint as this doesnt work between environments when I think about it.

svenngjeto commented:
Resolving in a phase independent should be possible via {{resolve:secretsmanager:UserCatalogueApiKey:SecretString}} in the Cloudformation template

janbj commented on deploy_nva_api.yml:
```diff
@@ -211,7 +211,7 @@ Resources:
       Parameters:
         CustomDomain: !Ref ApiCustomDomain
-        UserServiceSecretArn: !Sub 'arn:aws:secretsmanager:eu-west-1:${AWS::AccountId}:secret:UserCatalogueApiKey-dIqvz5'
+        UserServiceSecretArn: !Sub 'arn:aws:secretsmanager:eu-west-1:${AWS::AccountId}:secret:UserCatalogueApiKey-p9ebkD'
```
It just occurred to me that this ARN must be a template parameter because we have locked the template to one specific envirionment (where the ARN exists) with this change.

norrs replied:
@svenngjeto @janbj : Could anyone assist me on a videocall and we fix it to use the concept of `{{resolve:secretsmanager:UserCatalogueApiKey:SecretString}}` ? Poke me at slack.

norrs replied:
We have updated this @janbj , see new commits.

norrs committed:
Current nva-user-access-service requirements

We ideally want to refactor this and use AWS IAM (maybe together with
policies) to retrict which lambdas can communicate with other lambdas.

norrs committed:
Set default value for UserServiceApiSecret (name of secret)

norrs committed:
Merge branch 'develop' into update_user_catalogue_apikey_reference

janbj commented on deploy_nva_api.yml:
```diff
@@ -198,25 +213,26 @@ Resources:
         SemanticVersion: 0.1.2
       Parameters:
-        ApiKeySecretArn: !Sub 'arn:aws:secretsmanager:eu-west-1:${AWS::AccountId}:secret:UserCatalogueApiKey-dIqvz5'
+        ApiKeySecretArn: !Sub 'arn:aws:secretsmanager:eu-west-1:${AWS::AccountId}:secret:UserCatalogueApiKey-p9ebkD'
```
Will we not get in trouble when trying this template in the wrong environment?

norrs replied:
Overlooked, fixed by https://github.com/BIBSYSDEV/NVA-deploy/pull/13/commits/38218a987724ebf24122dbc9a05432cec489d861

norrs committed:
Use template parameter for UserServiceApiSecretArn

norrs committed:
Use new released cognito-pre-token-generation-trigger
